api/src/main/java/org/openmrs/module/addresshierarchy/config/AddressConfigurationLoader.java

```diff
@@ -0,0 +1,274 @@
+package org.openmrs.module.addresshierarchy.config;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.GlobalProperty;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.addresshierarchy.AddressHierarchyLevel;
+import org.openmrs.module.addresshierarchy.service.AddressHierarchyService;
+import org.openmrs.module.addresshierarchy.util.AddressHierarchyImportUtil;
+import org.openmrs.util.OpenmrsConstants;
+import org.openmrs.util.OpenmrsUtil;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.io.xml.DomDriver;
+
+/**
+ * Responsible for loading the address configuration appropriately
+ */
+public class AddressConfigurationLoader {
+
+	protected static final String ADDR_CONFIG_FILE_NAME = "addressConfiguration.xml";
+
+	public static final String NOT_COMPUTABLE_CHECKSUM = "not_computable_checksum";
+	public static final String NOT_READABLE_CHECKSUM = "not_readadble_checksum";
+
+	public static final String CHECKSUM_FILE_EXT = "checksum";
+
+	protected static Log log = LogFactory.getLog(AddressConfigurationLoader.class);
+
+	/**
+	 * @return The path to the configuration subdirectory.
+	 */
+	public static String getSubdirConfigPath() {
+		return new StringBuilder()
+				.append(OpenmrsUtil.getApplicationDataDirectory())
+				.append("configuration").append(File.separator).append("addresshierarchy")
+				.toString();
+	}
+
+	public static void loadAddressConfiguration() {
+
+		final ConfigLoaderUtil configUtil = new ConfigLoaderUtil(getSubdirConfigPath());
+
+		String xmlConfigFileName = ADDR_CONFIG_FILE_NAME;
+
+		File configFile = configUtil.getFile(xmlConfigFileName);
+		if (!configFile.exists()) {
+			log.warn("Address hierarchy configuration file appears invalid, skipping the loading process: " + xmlConfigFileName);
```

Comment:
I'd want to make this more explicit.  Since this method is always called every time the module starts, I wouldn't want a warning in my log file if I never intended to load configuration from a file.  ie. - if you will hit this code if you choose not to use this feature at all, then at most I would want debug-level logging.

If a config file is found, but it fails to load or is in the wrong format, then this is a case where you would want to fail hard, and actually stop the module from starting.  And it should be error-level logging.